File: examples/python/03_persian_whisper.py
"""Persian speech-to-text with Whisper, and text-to-speech back again.

Two things matter for Persian accuracy:

* Pass `language="fa"`. Without it Whisper guesses from the first seconds of
  audio and often mislabels Persian as Arabic or Urdu, which wrecks the output.
* Use `prompt=` to seed proper nouns, product names and spelling conventions.
  Whisper conditions on it and will spell those terms your way.

Audio is billed per second, not per token, so the Toman cost helper deliberately
does not price it -- see https://1xai.ir/pricing .

    python 03_persian_whisper.py voice.m4a
"""

import sys

from onexai import OneXAI

client = OneXAI()
path = sys.argv[1] if len(sys.argv) > 1 else "voice.m4a"

with open(path, "rb") as audio:
    transcript = client.audio.transcriptions.create(
        model="whisper-1",
        file=audio,
        language="fa",
        prompt="اصطلاحات فنی: کوبرنتیز، داکر، ری‌اکت، پایتون.",
    )

print(transcript.text)

# ...and speak it back. TTS is billed per character.
speech = client.audio.speech.create(
    model="tts-1",
    voice="alloy",
    input=transcript.text[:500],
)
speech.write_to_file("reply.mp3")
print("wrote reply.mp3")

File: examples/node/03_persian_whisper.mjs
/**
 * Persian speech-to-text with Whisper.
 *
 * Two things matter for Persian accuracy:
 *
 * - Pass `language: "fa"`. Without it Whisper guesses from the first seconds of
 *   audio and often mislabels Persian as Arabic or Urdu, wrecking the output.
 * - Use `prompt` to seed proper nouns and spelling conventions. Whisper
 *   conditions on it and will spell those terms your way.
 *
 * Audio is billed per second, not per token, so the Toman cost helper
 * deliberately does not price it -- see https://1xai.ir/pricing .
 *
 *   node 03_persian_whisper.mjs voice.m4a
 */

import { createReadStream } from "node:fs";
import { writeFile } from "node:fs/promises";

import { OneXAI } from "onexai";

const client = new OneXAI();
const path = process.argv[2] ?? "voice.m4a";

const transcript = await client.audio.transcriptions.create({
  model: "whisper-1",
  file: createReadStream(path),
  language: "fa",
  prompt: "اصطلاحات فنی: کوبرنتیز، داکر، ری‌اکت، پایتون.",
});

console.log(transcript.text);

// ...and speak it back. TTS is billed per character.
const speech = await client.audio.speech.create({
  model: "tts-1",
  voice: "alloy",
  input: transcript.text.slice(0, 500),
});

await writeFile("reply.mp3", Buffer.from(await speech.arrayBuffer()));
console.log("wrote reply.mp3");
